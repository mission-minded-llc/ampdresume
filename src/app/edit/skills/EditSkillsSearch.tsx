"use client";

import { useSession } from "next-auth/react";
import React, { useEffect, useMemo, useState } from "react";
import {
  Box,
  Button,
  Checkbox,
  Dialog,
  DialogActions,
  DialogContent,
  Divider,
  FormControlLabel,
  List,
  ListItem,
  ListItemIcon,
  ListItemText,
  Paper,
  TextField,
  Typography,
} from "@mui/material";
import { Icon } from "@iconify/react";
import { useMutation, useQuery, useQueryClient } from "@tanstack/react-query";
import { CustomDialogTitle } from "@/components/CustomDialogTitle";
import { IconSelector } from "@/components/IconSelector";
import { LoadingOverlay } from "@/components/LoadingOverlay";
import { Tooltip } from "@/components/Tooltip";
import { TooltipTotalYears } from "@/components/tooltips";
import { addSkill } from "@/graphql/addSkill";
import { addSkillForUser } from "@/graphql/addSkillForUser";
import { getSkills } from "@/graphql/getSkills";
import { removeLeadingZero } from "@/lib/format";

export const EditSkillsSearch = () => {
  const { data: session, status } = useSession();
  const queryClient = useQueryClient();

  const [isAddExistingSkillDialogOpen, setIsAddExistingSkillDialogOpen] = useState(false);
  const [isAddNewSkillDialogOpen, setIsAddNewSkillDialogOpen] = useState(false);
  const [newSkillName, setNewSkillName] = useState("");
  const [selectedSkillId, setSelectedSkillId] = useState<string | null>(null);
  const [yearStarted, setYearStarted] = useState(new Date().getFullYear());
  const [totalYears, setTotalYears] = useState(0);
  const [searchTerm, setSearchTerm] = useState("");
  const [icon, setIcon] = useState<string | null>(null);

  // Add state for auto-calculate
  const hasTotalYears = totalYears && totalYears > 0;
  const defaultAutoCalculate = !hasTotalYears;
  const [autoCalculate, setAutoCalculate] = useState(defaultAutoCalculate);

  // When the search term is at least this length, the search will trigger
  // and show the list of skills. Pressing "Esc" will clear the search term.
  const minCharsForSearch = 2;

  useEffect(() => {
    setNewSkillName(searchTerm);

    const handleKeyDown = (e: KeyboardEvent) => {
      if (e.key === "Escape") setSearchTerm("");
    };

    window.addEventListener("keydown", handleKeyDown);
    return () => {
      window.removeEventListener("keydown", handleKeyDown);
    };
  }, [searchTerm]);

  // Load all available skills for search/selector
  const { isPending, error, data } = useQuery({
    enabled: status === "authenticated" && !!session?.user?.id,
    queryKey: ["skills"],
    queryFn: async () => await getSkills(),
  });

  const addExistingSkillMutation = useMutation({
    mutationFn: async ({
      skillId,
      yearStarted,
      totalYears,
    }: {
      skillId: string;
      yearStarted: number;
      totalYears: number;
    }) => {
      if (!session?.user?.id) return;

      await addSkillForUser({
        userId: session.user.id,
        skillId,
        yearStarted,
        totalYears,
      });
    },
    onSuccess: () => {
      queryClient.invalidateQueries({ queryKey: ["skillsForUser"] });
    },
  });

  const addNewSkillMutation = useMutation({
    mutationFn: async ({ name, icon }: { name: string; icon: string }) => {
      if (!session?.user?.id) return;
      // addSkill now returns the new skill object with its ID
      return await addSkill({ userId: session.user.id, name, icon });
    },
    onSuccess: (data) => {
      // data is the new skill object
      if (data?.id) {
        setSelectedSkillId(data.id);
        setIsAddExistingSkillDialogOpen(true);
        setIsAddNewSkillDialogOpen(false);
        setNewSkillName("");
        setIcon(null);
      }
    },
  });

  // Filter skills based on search term
  const filteredSkills = useMemo(() => {
    if (!data?.skills || searchTerm.length < minCharsForSearch) return [];

    // Sort skills by closest match and limit to top 10
    return data.skills
      .filter((skill) => skill.name.toLowerCase().includes(searchTerm.toLowerCase()))
      .sort((a, b) => {
        if (a.name.toLowerCase() === searchTerm.toLowerCase()) return -1;
        if (b.name.toLowerCase() === searchTerm.toLowerCase()) return 1;

        // Basic sorting by how close the match is to the start of the name
        const aIndex = a.name.toLowerCase().indexOf(searchTerm.toLowerCase());
        const bIndex = b.name.toLowerCase().indexOf(searchTerm.toLowerCase());

        return aIndex - bIndex;
      })
      .slice(0, 10); // Limit to top 10 matches
  }, [data?.skills, searchTerm]);

  if (isPending) return <LoadingOverlay message="Loading skills..." />;
  if (error) return <Box>Error loading skills: {error.message}</Box>;

  const handleSkillSelection = (skillId: string) => {
    // Open dialog to get proficiency level
    setSelectedSkillId(skillId);
    setIsAddExistingSkillDialogOpen(true);
  };

  const handleAddExistingSkill = () => {
    if (selectedSkillId) {
      addExistingSkillMutation.mutate({
        skillId: selectedSkillId,
        yearStarted: yearStarted,
        totalYears: totalYears,
      });

      // Close dialog and reset states
      setIsAddExistingSkillDialogOpen(false);
      setSelectedSkillId(null);
      setYearStarted(new Date().getFullYear());
      setTotalYears(0);
      setSearchTerm("");
    }
  };

  const handleAddNewSkill = () => {
    // Only call mutate if icon is a string (not null)
    if (newSkillName && icon) {
      addNewSkillMutation.mutate({ name: newSkillName, icon });
    }
  };

  return (
    <Box>
      <TextField
        fullWidth
        variant="outlined"
        label="Search Skills to Add"
        value={searchTerm}
        onChange={(e) => setSearchTerm(e.target.value)}
        sx={{ mb: 2 }}
        name="searchSkills"
        placeholder="Type at least 3 characters to search"
      />
      {searchTerm.length >= minCharsForSearch && (
        <Paper elevation={3}>
          <List>
            {filteredSkills.length > 0 ? (
              filteredSkills.map((skill) => (
                <ListItem
                  key={skill.id}
                  sx={(theme) => ({
                    cursor: "pointer",
                    "&:hover": {
                      backgroundColor: theme.palette.action.hover,
                    },
                  })}
                  onClick={() => handleSkillSelection(skill.id)}
                >
                  <ListItemIcon>
                    {skill?.icon ? <Icon icon={skill.icon} width={24} height={24} /> : null}
                  </ListItemIcon>
                  <ListItemText primary={skill.name} />
                </ListItem>
              ))
            ) : (
              <ListItem>
                <ListItemText primary="No matching skills found" />
              </ListItem>
            )}
            <Divider />
            <ListItem
              sx={{ cursor: "pointer", backgroundColor: "action.hover" }}
              onClick={() => setIsAddNewSkillDialogOpen(true)}
            >
              <ListItemIcon>
                <Icon icon="mdi:plus" width={24} height={24} />
              </ListItemIcon>
              <ListItemText primary={`Add a new skill: ${searchTerm}`} />
            </ListItem>
          </List>
        </Paper>
      )}

      <Dialog
        open={isAddExistingSkillDialogOpen}
        onClose={() => setIsAddExistingSkillDialogOpen(false)}
        maxWidth="md"
      >
        <CustomDialogTitle closeHandler={() => setIsAddExistingSkillDialogOpen(false)}>
          Enter Proficiency Level
          <Tooltip message={<TooltipTotalYears />} />
        </CustomDialogTitle>

        <DialogContent>
          <Box
            sx={{
              display: "flex",
              flexDirection: "column",
              gap: 2,
              width: "100%",
            }}
          >
            <Box sx={{ display: "flex", gap: 2, width: "100%" }}>
              {autoCalculate ? (
                <TextField
                  autoFocus
                  margin="dense"
                  type="number"
                  fullWidth
                  variant="outlined"
                  value={yearStarted}
                  label="Year Started"
                  name="yearStarted"
                  onInput={(e) => {
                    const target = e.target as HTMLInputElement;
                    target.value = removeLeadingZero(target.value);
                  }}
                  onChange={(e) => setYearStarted(Number(e.target.value))}
                  slotProps={{
                    htmlInput: { min: 1900, max: new Date().getFullYear() },
                  }}
                />
              ) : (
                <TextField
                  margin="dense"
                  type="number"
                  fullWidth
                  variant="outlined"
                  value={totalYears}
                  label="Total Years"
                  name="totalYears"
                  onInput={(e) => {
                    const target = e.target as HTMLInputElement;
                    target.value = Math.max(
                      0,
                      Math.min(parseInt(removeLeadingZero(target.value)), 100),
                    ).toString();
                  }}
                  onChange={(e) => setTotalYears(Number(e.target.value))}
                  slotProps={{ htmlInput: { min: 0, max: 100 } }}
                />
              )}
            </Box>
            <Box sx={{ ml: 1, mt: 2, mb: 2 }}>
              <FormControlLabel
                control={
                  <Checkbox
                    checked={autoCalculate}
                    onChange={(event) => {
                      const checked = event.target.checked;
                      setAutoCalculate(checked);
                      if (checked) {
                        setTotalYears(0);
                      }
                    }}
                    name="autoCalculate"
                  />
                }
                label={
                  <Typography variant="body2" sx={{ fontSize: "0.8rem", fontStyle: "italic" }}>
                    Auto-calculate <strong>years of experience</strong> based on year started.
                  </Typography>
                }
              />
            </Box>
          </Box>
        </DialogContent>
        <DialogActions>
          <Button onClick={() => setIsAddExistingSkillDialogOpen(false)}>Cancel</Button>
          <Button onClick={handleAddExistingSkill}>Add Skill</Button>
        </DialogActions>
      </Dialog>

      <Dialog
        open={isAddNewSkillDialogOpen}
        onClose={() => setIsAddNewSkillDialogOpen(false)}
        maxWidth="lg"
      >
        <CustomDialogTitle closeHandler={() => setIsAddNewSkillDialogOpen(false)}>
          Add a New Skill
        </CustomDialogTitle>
        <DialogContent sx={{ minHeight: "300px" }}>
          <Box
            sx={{
              display: "flex",
              flexDirection: "column",
              gap: 2,
              width: "100%",
            }}
          >
            <TextField
              fullWidth
              variant="outlined"
              label="Skill Name"
              value={newSkillName}
              onChange={(e) => setNewSkillName(e.target.value)}
              sx={{ mt: 2 }}
            />
            <Box sx={{ display: "flex", gap: 2, width: "100%" }}>
              <IconSelector setIcon={setIcon} icon={icon} limit={20} />
            </Box>
          </Box>
        </DialogContent>
        <DialogActions>
          <Button onClick={() => setIsAddNewSkillDialogOpen(false)}>Cancel</Button>
          <Button onClick={handleAddNewSkill} disabled={!newSkillName || !icon}>
            Add Skill
          </Button>
        </DialogActions>
      </Dialog>
    </Box>
  );
};
