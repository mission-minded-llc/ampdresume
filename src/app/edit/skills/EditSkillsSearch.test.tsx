import "@testing-library/jest-dom";

import { fireEvent, render, waitFor } from "@testing-library/react";
import { useQuery, useQueryClient } from "@tanstack/react-query";

import { EditSkillsSearch } from "./EditSkillsSearch";
import React from "react";
import { useSession } from "next-auth/react";

jest.mock("next-auth/react", () => ({
  useSession: jest.fn(),
}));

jest.mock("@tanstack/react-query", () => ({
  useQuery: jest.fn(),
  useMutation: jest.fn(),
  useQueryClient: jest.fn(),
}));

jest.mock("@/graphql/getSkills", () => ({
  getSkills: jest.fn(),
}));

jest.mock("@/graphql/addSkillForUser", () => ({
  addSkillForUser: jest.fn(),
}));

jest.mock("@/components/CustomDialogTitle", () => ({
  CustomDialogTitle: ({
    children,
    closeHandler,
  }: {
    children: React.ReactNode;
    closeHandler: () => void;
  }) => (
    <div>
      <button onClick={closeHandler}>Close</button>
      {children}
    </div>
  ),
}));

jest.mock("@/components/Tooltip", () => ({
  Tooltip: ({ message }: { message: React.ReactNode }) => <div>{message}</div>,
}));

jest.mock("@/components/tooltips", () => ({
  TooltipTotalYears: () => <div>TooltipTotalYears</div>,
}));

jest.mock("@/components/LoadingOverlay", () => ({
  LoadingOverlay: ({ message }: { message: string }) => <div>{message}</div>,
}));

describe("EditSkillsSearch", () => {
  const mockSession = {
    user: { id: "user-id" },
  };

  const mockQueryClient = {
    invalidateQueries: jest.fn(),
  };

  beforeEach(() => {
    jest.clearAllMocks();
    (useSession as jest.Mock).mockReturnValue({ data: mockSession, status: "authenticated" });
    (useQueryClient as jest.Mock).mockReturnValue(mockQueryClient);
  });

  it("renders correctly", () => {
    (useQuery as jest.Mock).mockReturnValue({ isPending: false, data: { skills: [] } });

    const { container, getByLabelText } = render(<EditSkillsSearch />);
    expect(getByLabelText("Search Skills to Add")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it("displays loading overlay when skills are being fetched", () => {
    (useQuery as jest.Mock).mockReturnValue({ isPending: true });

    const { container, getByText } = render(<EditSkillsSearch />);
    expect(getByText("Loading skills...")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it("displays error message when skills fetching fails", () => {
    (useQuery as jest.Mock).mockReturnValue({ isPending: false, error: { message: "Error" } });

    const { container, getByText } = render(<EditSkillsSearch />);
    expect(getByText("Error loading skills: Error")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it("filters and displays skills based on search term", async () => {
    const mockSkills = [
      { id: "1", name: "JavaScript", icon: "icon-js" },
      { id: "2", name: "TypeScript", icon: "icon-ts" },
    ];
    (useQuery as jest.Mock).mockReturnValue({ isPending: false, data: { skills: mockSkills } });

    const { container, getByLabelText, getByText } = render(<EditSkillsSearch />);
    const searchInput = getByLabelText("Search Skills to Add");
    expect(container).toMatchSnapshot();

    fireEvent.change(searchInput, { target: { value: "Java" } });

    await waitFor(() => {
      expect(getByText("JavaScript")).toBeInTheDocument();
      expect(container).toMatchSnapshot();
    });

    fireEvent.change(searchInput, { target: { value: "Java" } });

    await waitFor(() => {
      expect(getByText("JavaScript")).toBeInTheDocument();
      expect(container).toMatchSnapshot();
    });
  });

  it("opens dialog when a skill is selected", async () => {
    const mockSkills = [{ id: "1", name: "JavaScript", icon: "icon-js" }];
    (useQuery as jest.Mock).mockReturnValue({ isPending: false, data: { skills: mockSkills } });

    const { container, getByLabelText, getByText } = render(<EditSkillsSearch />);
    const searchInput = getByLabelText("Search Skills to Add");
    expect(container).toMatchSnapshot();

    fireEvent.change(searchInput, { target: { value: "Java" } });

    await waitFor(() => {
      expect(getByText("JavaScript")).toBeInTheDocument();
      expect(container).toMatchSnapshot();
    });

    fireEvent.click(getByText("JavaScript"));

    expect(getByText("Enter Proficiency Level")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
