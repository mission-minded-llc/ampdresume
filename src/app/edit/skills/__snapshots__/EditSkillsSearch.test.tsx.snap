// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`EditSkillsSearch displays error message when skills fetching fails 1`] = `
<div>
  <div
    class="MuiBox-root css-0"
  >
    Error loading skills: 
    Error
  </div>
</div>
`;

exports[`EditSkillsSearch displays loading overlay when skills are being fetched 1`] = `
<div>
  <div>
    Loading skills...
  </div>
</div>
`;

exports[`EditSkillsSearch filters and displays skills based on search term 1`] = `
<div>
  <div
    class="MuiBox-root css-0"
  >
    <div
      class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-dzmwfx-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-19qnlrw-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="false"
        for="«r2»"
        id="«r2»-label"
      >
        Search Skills to Add
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1blp12k-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input css-16wblaj-MuiInputBase-input-MuiOutlinedInput-input"
          id="«r2»"
          name="searchSkills"
          placeholder="Type at least 3 characters to search"
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-18p5xg2-MuiNotchedOutlined-root-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-1n64csd-MuiNotchedOutlined-root"
          >
            <span>
              Search Skills to Add
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
</div>
`;

exports[`EditSkillsSearch filters and displays skills based on search term 2`] = `
<div>
  <div
    class="MuiBox-root css-0"
  >
    <div
      class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-dzmwfx-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-113d811-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="true"
        for="«r2»"
        id="«r2»-label"
      >
        Search Skills to Add
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1blp12k-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input css-16wblaj-MuiInputBase-input-MuiOutlinedInput-input"
          id="«r2»"
          name="searchSkills"
          placeholder="Type at least 3 characters to search"
          type="text"
          value="Java"
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-18p5xg2-MuiNotchedOutlined-root-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-ex8a5f-MuiNotchedOutlined-root"
          >
            <span>
              Search Skills to Add
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 css-1gtchvp-MuiPaper-root"
      style="--Paper-shadow: 0px 3px 3px -2px rgba(0,0,0,0.2),0px 3px 4px 0px rgba(0,0,0,0.14),0px 1px 8px 0px rgba(0,0,0,0.12);"
    >
      <ul
        class="MuiList-root MuiList-padding css-b7dj7n-MuiList-root"
      >
        <li
          class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-gys9a0-MuiListItem-root"
        >
          <div
            class="MuiListItemIcon-root css-cokf1l-MuiListItemIcon-root"
          >
            <span
              data-icon="icon-js"
              data-testid="icon"
              height="24"
              width="24"
            />
          </div>
          <div
            class="MuiListItemText-root css-cfq8qh-MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-rizt0-MuiTypography-root"
            >
              JavaScript
            </span>
          </div>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`EditSkillsSearch filters and displays skills based on search term 3`] = `
<div>
  <div
    class="MuiBox-root css-0"
  >
    <div
      class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-dzmwfx-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-113d811-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="true"
        for="«r2»"
        id="«r2»-label"
      >
        Search Skills to Add
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1blp12k-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input css-16wblaj-MuiInputBase-input-MuiOutlinedInput-input"
          id="«r2»"
          name="searchSkills"
          placeholder="Type at least 3 characters to search"
          type="text"
          value="Java"
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-18p5xg2-MuiNotchedOutlined-root-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-ex8a5f-MuiNotchedOutlined-root"
          >
            <span>
              Search Skills to Add
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 css-1gtchvp-MuiPaper-root"
      style="--Paper-shadow: 0px 3px 3px -2px rgba(0,0,0,0.2),0px 3px 4px 0px rgba(0,0,0,0.14),0px 1px 8px 0px rgba(0,0,0,0.12);"
    >
      <ul
        class="MuiList-root MuiList-padding css-b7dj7n-MuiList-root"
      >
        <li
          class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-gys9a0-MuiListItem-root"
        >
          <div
            class="MuiListItemIcon-root css-cokf1l-MuiListItemIcon-root"
          >
            <span
              data-icon="icon-js"
              data-testid="icon"
              height="24"
              width="24"
            />
          </div>
          <div
            class="MuiListItemText-root css-cfq8qh-MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-rizt0-MuiTypography-root"
            >
              JavaScript
            </span>
          </div>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`EditSkillsSearch opens dialog when a skill is selected 1`] = `
<div>
  <div
    class="MuiBox-root css-0"
  >
    <div
      class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-dzmwfx-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-19qnlrw-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="false"
        for="«r4»"
        id="«r4»-label"
      >
        Search Skills to Add
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1blp12k-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input css-16wblaj-MuiInputBase-input-MuiOutlinedInput-input"
          id="«r4»"
          name="searchSkills"
          placeholder="Type at least 3 characters to search"
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-18p5xg2-MuiNotchedOutlined-root-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-1n64csd-MuiNotchedOutlined-root"
          >
            <span>
              Search Skills to Add
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
</div>
`;

exports[`EditSkillsSearch opens dialog when a skill is selected 2`] = `
<div>
  <div
    class="MuiBox-root css-0"
  >
    <div
      class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-dzmwfx-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-113d811-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="true"
        for="«r4»"
        id="«r4»-label"
      >
        Search Skills to Add
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1blp12k-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input css-16wblaj-MuiInputBase-input-MuiOutlinedInput-input"
          id="«r4»"
          name="searchSkills"
          placeholder="Type at least 3 characters to search"
          type="text"
          value="Java"
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-18p5xg2-MuiNotchedOutlined-root-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-ex8a5f-MuiNotchedOutlined-root"
          >
            <span>
              Search Skills to Add
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 css-1gtchvp-MuiPaper-root"
      style="--Paper-shadow: 0px 3px 3px -2px rgba(0,0,0,0.2),0px 3px 4px 0px rgba(0,0,0,0.14),0px 1px 8px 0px rgba(0,0,0,0.12);"
    >
      <ul
        class="MuiList-root MuiList-padding css-b7dj7n-MuiList-root"
      >
        <li
          class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-gys9a0-MuiListItem-root"
        >
          <div
            class="MuiListItemIcon-root css-cokf1l-MuiListItemIcon-root"
          >
            <span
              data-icon="icon-js"
              data-testid="icon"
              height="24"
              width="24"
            />
          </div>
          <div
            class="MuiListItemText-root css-cfq8qh-MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-rizt0-MuiTypography-root"
            >
              JavaScript
            </span>
          </div>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`EditSkillsSearch opens dialog when a skill is selected 3`] = `
<div
  aria-hidden="true"
>
  <div
    class="MuiBox-root css-0"
  >
    <div
      class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-dzmwfx-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-113d811-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="true"
        for="«r4»"
        id="«r4»-label"
      >
        Search Skills to Add
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1blp12k-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input css-16wblaj-MuiInputBase-input-MuiOutlinedInput-input"
          id="«r4»"
          name="searchSkills"
          placeholder="Type at least 3 characters to search"
          type="text"
          value="Java"
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-18p5xg2-MuiNotchedOutlined-root-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-ex8a5f-MuiNotchedOutlined-root"
          >
            <span>
              Search Skills to Add
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 css-1gtchvp-MuiPaper-root"
      style="--Paper-shadow: 0px 3px 3px -2px rgba(0,0,0,0.2),0px 3px 4px 0px rgba(0,0,0,0.14),0px 1px 8px 0px rgba(0,0,0,0.12);"
    >
      <ul
        class="MuiList-root MuiList-padding css-b7dj7n-MuiList-root"
      >
        <li
          class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-gys9a0-MuiListItem-root"
        >
          <div
            class="MuiListItemIcon-root css-cokf1l-MuiListItemIcon-root"
          >
            <span
              data-icon="icon-js"
              data-testid="icon"
              height="24"
              width="24"
            />
          </div>
          <div
            class="MuiListItemText-root css-cfq8qh-MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-rizt0-MuiTypography-root"
            >
              JavaScript
            </span>
          </div>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`EditSkillsSearch renders correctly 1`] = `
<div>
  <div
    class="MuiBox-root css-0"
  >
    <div
      class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-dzmwfx-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-19qnlrw-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="false"
        for="«r0»"
        id="«r0»-label"
      >
        Search Skills to Add
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1blp12k-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input css-16wblaj-MuiInputBase-input-MuiOutlinedInput-input"
          id="«r0»"
          name="searchSkills"
          placeholder="Type at least 3 characters to search"
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-18p5xg2-MuiNotchedOutlined-root-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-1n64csd-MuiNotchedOutlined-root"
          >
            <span>
              Search Skills to Add
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
</div>
`;
